### PR TITLE
Consider all cases in path generation

### PR DESCRIPTION
**Pull request by davidkpiano**
CleanShot 2022-05-26 at 11 25 56@2x

### CODE REVIEW

File: packages/xstate-test/test/dieHard.test.ts [hunk 1]
```diff
@@ -270,10 +270,10 @@ describe('error path trace', () => {
               	State: {\\"value\\":\\"first\\"}
               	Event: {\\"type\\":\\"NEXT\\"}
 
-              	State: {\\"value\\":\\"second\\"} via NEXT
+              	State: {\\"value\\":\\"second\\"} via {\\"type\\":\\"NEXT\\"}
               	Event: {\\"type\\":\\"NEXT\\"}
 
-              	State: {\\"value\\":\\"third\\"} via NEXT"
```
Comment (davidkpiano) [inline packages/xstate-test/test/dieHard.test.ts:276]: Not sure why this disappears; investigating

File: packages/xstate-test/src/machine.ts [hunk 2]
```diff
@@ -58,15 +58,13 @@ function serializeMachineTransition(
   event: AnyEventObject | null,
   { serializeEvent }: { serializeEvent: (event: AnyEventObject) => string }
 ): string {
-  if (!event) {
+  // Only consider the transition via the serialized event if there actually
+  // was a defined transition for the event
+  if (!event || state.transitions.length === 0) {
```
[the remark this answers is not in the document]
Comment (davidkpiano) [reply packages/xstate-test/src/machine.ts:63]: The test is this one actually; https://github.com/statelyai/xstate/pull/3343/files#diff-d7468424c510a8f0d4f4fcf0d2cbfe0ecda06e489f98982f501eb4275698a51eR187

[the remark this answers is not in the document]
Comment (Andarist) [reply packages/xstate-test/src/machine.ts:63]: ah OK 👍  so this commit is only simplifying logic but without any behavior changes?

[the remark this answers is not in the document]
Comment (davidkpiano) [reply packages/xstate-test/src/machine.ts:63]: Added a visual to the test description

**Review by mattpocock — APPROVED** [review]

**Pull request merged**